File: frontend/my-app/types/facility.ts
export interface StarRatings {
  overall: string | null;
  health_inspection: string | null;
  staffing: string | null;
  quality_care: string | null;
}

export interface ClaimsMetrics {
  short_term_hospitalization: string | null;
  str_national_avg_hospitalization: string | null;
  str_state_avg_hospitalization: string | null;
  str_ed_visit: string | null;
  str_ed_visits_national_avg: string | null;
  str_ed_visits_state_avg: string | null;
  lt_hospitalization: string | null;
  lt_national_avg_hospitalization: string | null;
  lt_state_avg_hospitalization: string | null;
  lt_ed_visit: string | null;
  lt_ed_visits_national_avg: string | null;
  lt_ed_visits_state_avg: string | null;
}

export interface FacilityData {
  name_of_facility: string;
  location: string;
  state_code: string;
  census_capacity: string;
  star_ratings: StarRatings;
  claims_metrics: ClaimsMetrics;
}

export interface ManualInputs {
  nameOverride: string;
  emr: string;
  currentCensus: string;
  patientType: string;
  prevCoverage: string;
  prevPerformance: string;
  medicalCoverage: string;
}

File: backend/models/facility.py
from pydantic import BaseModel
from typing import Optional


class StarRatings(BaseModel):
    overall: Optional[str]
    health_inspection: Optional[str]
    staffing: Optional[str]
    quality_care: Optional[str]


class ClaimsMetrics(BaseModel):
    short_term_hospitalization: Optional[str]
    str_national_avg_hospitalization: Optional[str]
    str_state_avg_hospitalization: Optional[str]
    str_ed_visit: Optional[str]
    str_ed_visits_national_avg: Optional[str]
    str_ed_visits_state_avg: Optional[str]
    lt_hospitalization: Optional[str]
    lt_national_avg_hospitalization: Optional[str]
    lt_state_avg_hospitalization: Optional[str]
    lt_ed_visit: Optional[str]
    lt_ed_visits_national_avg: Optional[str]
    lt_ed_visits_state_avg: Optional[str]


class FacilityResponse(BaseModel):
    name_of_facility: Optional[str]
    location: str
    state_code: str
    census_capacity: Optional[str]
    star_ratings: StarRatings
    claims_metrics: ClaimsMetrics
